src/context.ts

```diff
@@ -68,21 +68,12 @@ function createContext<S extends StoreApi<unknown>>() {
         'Seems like you have not used zustand provider as an ancestor.'
       )
     }
-    return useMemo(
-      () =>
-        ({
-          getState: store.getState,
-          setState: store.setState,
-          subscribe: store.subscribe,
-          destroy: store.destroy,
-        } as WithoutCallSignature<S>),
-      [store]
-    )
+    return useMemo<WithoutCallSignature<S>>(() => ({ ...store }), [store])
```

Comment:
If you want to return all the props in the store, would be better just return the store directly

```suggestion
    return useMemo<WithoutCallSignature<S>>(() => store, [store])
```